### DISCUSSION TITLE
Can ragna return an answer with link to reference or source/context?

### DISCUSSION THREAD

**Discussion by asmith26** [Quansight/ragna#334] [Q&A] [answered]
Hi, just wondering can ragna return an answer with link to reference or source/context, possibly something like:

![image](https://github.com/phidatahq/ai-cookbook/assets/6988036/f4f7db8a-8817-4e35-83d2-430c65411bcb)

Thanks for any help!

**Comment by peachkeel** [comment, marked as answer]
I'm pretty sure the functionality you're looking for is in v0.2.0a3.

See also: https://github.com/Quansight/ragna/commit/3526e4cc6fe3d3385dc926ee17a24107d10644ff

You can upgrade as follows:

```sh
$ pip install --upgrade --pre 'ragna[all]'
```

**Reply by pmeier** [reply]
This will come as feature in the upcoming `0.2.0` release. As mentioned, you can install the alpha for the release already. The release tracker, on which we are going to announce the release candidate versions, is [here](https://github.com/Quansight/ragna/discussions/306).

**Reply by asmith26** [reply]
Thanks very much for the answers @peachkeel @pmeier - is there a way I can use a local LLM (e.g. integrating via https://github.com/ollama/ollama)? Thanks again!

**Reply by pmeier** [reply]
Yes, that is totally doable. The `0.3.0` release will focus on how to use local LLMs either directly or through another library like Ollama.
